ui/apps/platform/cypress/helpers/compliance.js

```diff
@@ -0,0 +1,90 @@
+import * as api from '../constants/apiEndpoints';
+import { headingPlural, selectors, url } from '../constants/CompliancePage';
+
+import { visit } from './visit';
+
+const opnamesForDashboard = [
+    'clustersCount',
+    'namespacesCount',
+    'nodesCount',
+    'deploymentsCount',
+    'runStatuses',
+    'getComplianceStandards',
+];
+
+const standardNames = [
+    'CIS Docker v1.2.0',
+    'CIS Kubernetes v1.5',
+    'HIPAA 164',
+    'NIST SP 800-190',
+    'NIST SP 800-53',
+    'PCI DSS 3.2.1',
+];
+
+export function visitComplianceDashboard() {
+    opnamesForDashboard.forEach((opname) => {
+        cy.intercept('POST', api.graphql(opname)).as(opname);
+    });
+    // Intercept requests for compliance standards, which have same opname but different value in payload.
+    cy.intercept('POST', api.graphql('complianceStandards'), (req) => {
+        const { where } = req.body.variables;
+        const alias = standardNames.find((standardName) => where === `Standard:${standardName}`);
+        if (typeof alias === 'string') {
+            req.alias = alias;
+        }
+    });
+
+    visit(url.dashboard);
+
+    cy.wait(opnamesForDashboard.map((opname) => `@${opname}`));
+    cy.wait(standardNames.map((standardName) => `@${standardName}`));
+    cy.get('h1:contains("Compliance")');
+}
+
+/*
+ * Assume location is compliance dashboard.
+ */
+export function scanCompliance() {
+    cy.intercept('POST', api.graphql('triggerScan')).as('triggerScan');
+    opnamesForDashboard.forEach((opname) => {
+        cy.intercept('POST', api.graphql(opname)).as(opname);
+    });
+
+    cy.get(selectors.scanButton).click().should('have.attr', 'disabled');
```

Comment:
Savvy review comment. Yeah, it was not clear to me the first (or second) time I read it.

Changed to separate calls:

```js
    cy.get(selectors.scanButton).should('not.have.attr', 'disabled');
    cy.get(selectors.scanButton).click();
    cy.get(selectors.scanButton).should('have.attr', 'disabled');
```

DOM assertions retry (another reason to break chains) so a slight delay to become disabled is not a problem, but it would be a problem if the button could become re-enabled too soon. For example, mock request.